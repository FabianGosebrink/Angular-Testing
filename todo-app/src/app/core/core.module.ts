import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';

import { TodoService } from './todo.service';

@NgModule({
  imports: [CommonModule],
  providers: [TodoService]
})
export class CoreModule {}
