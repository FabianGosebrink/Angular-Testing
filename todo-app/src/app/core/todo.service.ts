import { Injectable } from '@angular/core';
import { Observable } from 'rxjs/Observable';

import { Todo } from '../models/todo.models';
import { Observer } from 'rxjs/Observer';

@Injectable()
export class TodoService {
  private existingTodos: Todo[] = [];

  constructor() {}

  getAllTodos(): Observable<Todo[]> {
    return Observable.create((observer: Observer<Todo[]>) => {
      setTimeout(() => {
        observer.next(this.existingTodos);
        observer.complete();
      }, 200);
    });
  }

  getSingleTodo(id: string): Observable<Todo> {
    return Observable.create((observer: Observer<Todo>) => {
      setTimeout(() => {
        observer.next(this.existingTodos.find(x => x.id === id));
        observer.complete();
      }, 200);
    });
  }

  addTodo(description: string): Todo {
    const toAdd: Todo = new Todo();
    toAdd.created = new Date();
    toAdd.id = this.guid();
    toAdd.done = false;
    toAdd.description = description;
    this.existingTodos.push(toAdd);
    return toAdd;
  }

  updateTodo(toUpdate: Todo): Todo {
    this.existingTodos.map(obj =>
      this.existingTodos.find(o => o.id === obj.id)
    );
    return toUpdate;
  }

  deleteTodo(id: string) {
    this.existingTodos = this.existingTodos.filter(item => item.id !== id);
  }

  private guid() {
    function s4() {
      return Math.floor((1 + Math.random()) * 0x10000)
        .toString(16)
        .substring(1);
    }
    return (
      s4() +
      s4() +
      '-' +
      s4() +
      '-' +
      s4() +
      '-' +
      s4() +
      '-' +
      s4() +
      s4() +
      s4()
    );
  }
}
