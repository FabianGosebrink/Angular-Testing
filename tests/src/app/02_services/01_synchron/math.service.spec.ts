import { TestBed } from '@angular/core/testing';

import { MathService } from './math.service';

describe('MathService', () => {
  let service: MathService;

  beforeEach(() => {
    TestBed.configureTestingModule({});
    service = TestBed.inject(MathService);
  });

  it('should be created', () => {
    expect(service).toBeTruthy();
  });

  describe('add', () => {
    it('should add two numbers', () => {
      // arrange
      const first = 5;
      const second = 11;
      const expected = 16;

      // act
      const result = service.add(first, second);

      // assert
      expect(result).toBe(expected);
    });
  });

  describe('add', () => {
    [
      {
        first: 1,
        second: 2,
        expected: 3,
      },
      {
        first: 3,
        second: 4,
        expected: 7,
      },
      {
        first: 23,
        second: 11,
        expected: 34,
      },
    ].forEach(({ first, second, expected }) => {
      it(`should return ${expected} when adding ${first} and ${second}`, () => {
        // act
        const result = service.add(first, second);

        // assert
        expect(result).toBe(expected);
      });
    });
  });

  describe('add with snapshots', () => {
    [
      {
        first: 1,
        second: 2,
      },
      {
        first: 3,
        second: 4,
      },
      {
        first: 23,
        second: 11,
      },
    ].forEach(({ first, second }) => {
      it(`should add ${first} and ${second}`, () => {
        // act
        const result = service.add(first, second);

        // assert
        expect(result).toMatchSnapshot();
      });
    });
  });
});
