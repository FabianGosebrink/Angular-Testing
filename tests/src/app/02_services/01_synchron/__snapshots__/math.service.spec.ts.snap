// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MathService add with snapshots should add 1 and 2 1`] = `3`;

exports[`MathService add with snapshots should add 3 and 4 1`] = `7`;

exports[`MathService add with snapshots should add 23 and 11 1`] = `34`;
